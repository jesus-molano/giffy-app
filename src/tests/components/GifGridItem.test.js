import { shallow } from 'enzyme';
import GifGridItem from '../../Components/GifGridItem';


describe('GifGridItem component tests', () => {
  const title = 'Un Titulo';
	const url = 'https://localhost/testPage.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('should render the component correctly', () => {
		expect(wrapper).toMatchSnapshot();
  });
  test('should be a paragraph with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  })
  test('should be a <img> with alt and src from props', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  })
  test('should have "animate__fadeIn" animation', () => {
    const div = wrapper.find('div');
    expect(div.prop('className')).toContain('animate__fadeIn');

  })
});
