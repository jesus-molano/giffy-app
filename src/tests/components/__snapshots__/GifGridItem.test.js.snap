// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem component tests should render the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https://localhost/testPage.jpg"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
